### ISSUE TITLE
Bug: Incorrect stats vs. real file names in RealContentHashPlugin

### ISSUE THREAD

**Issue by koconnorIXL** [FormidableLabs/webpack-stats-plugin#56]
I encountered some unexpected errors today when I attempted to upgrade my webpack project to webpack v5. While I am not sure that the "fault" for these errors lies with webpack-stats-plugin, it seems to me like this plugin was certainly involved. So I wanted to post here in case anyone is interested, or can provide me with advice. Without further ado, here goes.

Specifically, the problem I was running into was that the JS filenames in my outputted stats.json file did not match the actual filenames of the files created by webpack. For example, I might see a file in stats.json called 7953/7953.c47fc061af5c71d99c89.bundle.js, but then find in my build output directory that this file does not exist. Instead, a file with a different hash appears: 7953/7953.d1d2ef05ee132273a495.bundle.js

I dug in a bit, and I am observing the following facts:

1) Both this plugin (webpack-stats-plugin) and RealContentHashPlugin (which webpack includes by default) are tapping into the processAssets compilation hook

https://github.com/FormidableLabs/webpack-stats-plugin/blob/main/lib/stats-writer-plugin.js#L79
https://github.com/webpack/webpack/blob/bbe12305d66c8ee2471a72b2806d94355511cc92/lib/optimize/RealContentHashPlugin.js#L69

2) This plugin seems to be tapped first, while RealContentHashPlugin runs second

3) When RealContentHashPlugin does run, it modifies the hash values associated to some assets. This means that most assets end up with a different hash compared to what was computed by webpack-stats-plugin. webpack-stats-plugin is never reinvoked to pick up these changes and emit new stats

Now I do have workarounds here. For example, I could force webpack to stop its default inclusion of RealContentHashPlugin by adding a [directive](https://webpack.js.org/configuration/optimization/#optimizationrealcontenthash) to my webpack configuration. I could also eschew this plugin and use the --json flag on the webpack CLI instead to output stats json at the end of the compilation process.

However, I figured I would open this bug because I do enjoy using this plugin, and it would be nice if I could continue to do so with RealContentHashPlugin, without having to do too many gymnastics.

Please let me know if I can provide more information.

**Comment by ryan-roemer** [comment]
Thanks for the detailed comment and info! One thing that would be awesome to help us jump in is a minimal repository with install and build steps outlined as well as what you would want to happen that is not from the build output.

We just updated the lifecycle hook we use to add the stats output asset and it’s much earlier than the now deprecated metros we used before. There are also different stages we can experiment with within the new hook, so hopefully we can confirm the issue and then move to fix it. Thanks!

**Comment by koconnorIXL** [comment]
Sure thing. Check out this repo: https://github.com/koconnorIXL/webpack-stats-sandbox

Run `npm install` and `npm run build`, then observe that the files listed in `dist/stats.json` do not match the filenames of the files in `dist/`

**Comment by ryan-roemer** [comment]
Thanks -- I can confirm the hashes are different in stats object vs. filename.

**Comment by ryan-roemer** [comment]
@koconnorIXL -- I have new commits in [`bug/webpack5`](https://github.com/FormidableLabs/webpack-stats-plugin/tree/bug/webpack5) / https://github.com/FormidableLabs/webpack-stats-plugin/pull/58 that I believe fix this issue. Can you confirm if you have a moment? 

Here's my your repository with the new branch-based plugin for me:

```
$ yarn build && tree dist && cat dist/stats.json | grep bundle.js
yarn run v1.22.5
$ webpack --mode production
assets by status 2.57 KiB [cached] 3 assets
asset stats.json 1.21 KiB [compared for emit]
runtime modules 718 bytes 3 modules
cacheable modules 4.48 KiB
  modules by path ./src/ 717 bytes
    ./src/a.js 17 bytes [built] [code generated]
    ./src/b.js 45 bytes [built] [code generated]
    ./src/c.js 45 bytes [built] [code generated]
    ./src/a.css 610 bytes [built] [code generated]
  modules by path ./node_modules/css-loader/dist/runtime/*.js 3.78 KiB
    ./node_modules/css-loader/dist/runtime/cssWithMappingToString.js 2.21 KiB [built] [code generated]
    ./node_modules/css-loader/dist/runtime/api.js 1.57 KiB [built] [code generated]
webpack 5.1.3 compiled successfully in 1870 ms

✨  Done in 4.12s.
dist
├── a
│   └── a.4a762f86dd9564050d41.bundle.js
├── b
│   └── b.8a9a73e7ad295eb54146.bundle.js
├── c
│   └── c.f854f32203c6dc5a3388.bundle.js
└── stats.json

3 directories, 4 files
          "name": "a/a.4a762f86dd9564050d41.bundle.js"
          "name": "b/b.8a9a73e7ad295eb54146.bundle.js"
          "name": "c/c.f854f32203c6dc5a3388.bundle.js"
```

**Comment by ryan-roemer** [comment]
Fixes released in `webpack-stats-plugin@1.0.1`. I have a regression test for this case now, so I'm pretty sure it's fixed.

**Comment by koconnorIXL** [comment]
Yep, things are looking good on my end. Thanks a lot!
